File: android/src/main/java/org/maplibre/reactnative/components/layers/types/line/MLRNLineLayerManager.java
package org.maplibre.reactnative.components.layers.types.line;

import com.facebook.react.uimanager.ThemedReactContext;

import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager;

public class MLRNLineLayerManager extends MLRNVectorLayerManager<MLRNLineLayer> {
    public static final String REACT_CLASS = "MLRNLineLayer";

    @Override
    public String getName() {
        return REACT_CLASS;
    }

    @Override
    protected MLRNLineLayer createLayerViewInstance(ThemedReactContext reactContext) {
        return new MLRNLineLayer(reactContext);
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/layers/types/line/MLRNLineLayerManager.kt
package org.maplibre.reactnative.components.layers.types.line

import com.facebook.react.uimanager.ThemedReactContext
import org.maplibre.reactnative.components.layers.MLRNVectorLayerManager

class MLRNLineLayerManager : MLRNVectorLayerManager<MLRNLineLayer?>() {
    override fun getName(): String {
        return REACT_CLASS
    }

    override fun createLayerViewInstance(reactContext: ThemedReactContext?): MLRNLineLayer {
        return MLRNLineLayer(reactContext)
    }

    companion object {
        const val REACT_CLASS: String = "MLRNLineLayer"
    }
}
